src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -76,16 +76,19 @@ private static boolean isNotSameRequester(ConsentArtefact consentDetail, String
                 !consentDetail.getPatient().getId().equals(requesterId);
     }
 
-    public Mono<String> askForConsent(RequestedDetail requestedDetail) {
-        final String requestId = UUID.randomUUID().toString();
-        return validatePatient(requestedDetail.getPatient().getId())
-                .then(validateHIPAndHIU(requestedDetail))
-                .then(saveRequest(requestedDetail, requestId))
-                .then(postConsentRequest.broadcastConsentRequestNotification(ConsentRequest.builder()
-                        .detail(requestedDetail)
-                        .id(requestId)
-                        .build()))
-                .thenReturn(requestId);
+    public Mono<String> askForConsent(RequestedDetail requestedDetail, String requestId) {
+        final String consentRequestId = UUID.randomUUID().toString();
```

Comment:
```suggestion
        final String consentRequestId = UUID.randomUUID().toString();
```
Do we need this anymore?

Reply:
Thats the consentRequestId, so we need it. We save it to DB also broadcast it.

Reply:
I was just wondering why can't we use the unique id sent by the HIU as a consent request id?